### PR TITLE
Hubs/Scopes Merge 42d - Close previous scopes before binding a new global client

### PR DESCRIPTION
**Pull request by adinauer**
## :bulb: Motivation and Context

When calling `close` on the previous scopes after a new client has already been bound to the global scope, the new client is closed and Sentry does not capture events anymore.

## :pencil: Checklist

- [ ] I reviewed the submitted code.
- [ ] I added tests to verify the changes.
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [ ] I updated the docs if needed.
- [ ] Review from the native team if needed.
- [ ] No breaking change or entry added to the changelog.
- [ ] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

File: sentry/src/test/java/io/sentry/metrics/MetricsIntegrationTest.kt [hunk 1]
```diff
@@ -11,10 +11,16 @@ import org.mockito.kotlin.check
 import org.mockito.kotlin.mock
 import org.mockito.kotlin.verify
 import org.mockito.kotlin.whenever
+import kotlin.test.BeforeTest
 import kotlin.test.Test
 
 class MetricsIntegrationTest {
 
+    @BeforeTest
```
Comment (adinauer) [inline sentry/src/test/java/io/sentry/metrics/MetricsIntegrationTest.kt:19]: Test started becoming flaky. Not sure why exactly.

File: sentry/src/test/java/io/sentry/SentryTest.kt [hunk 2]
```diff
@@ -76,6 +76,38 @@ class SentryTest {
...
+
+        Sentry.init {
+            it.dsn = dsn
+        }
+        Sentry.setCurrentScopes(scopes)
+        Sentry.init {
+            it.dsn = dsn
+        }
+        verify(scopes).close(eq(true))
+        assertTrue(Sentry.getGlobalScope().client.isEnabled)
```
Comment (lbloder) [inline sentry/src/test/java/io/sentry/SentryTest.kt:93]: Should we also check that `Sentry.getCurrentScopes()` has an enabled client?

**Review by lbloder — APPROVED** [review]
LGTM 👍 
One question on test

File: sentry/src/test/java/io/sentry/SentryTest.kt (hunk 2 above)
Comment (adinauer) [reply sentry/src/test/java/io/sentry/SentryTest.kt:93]: We basically already are. It'd be more of a duplicate of tests already present in ScopesTest / CombinedScopeViewTest. I think it's fine as is.

**Pull request merged**